### PR TITLE
Removes ghost check for battlecruiser objective and fixes players being able to roll it multiple times in a row

### PR DESCRIPTION
**Pull request by Salex08**
## About The Pull Request

Removes the ghost check for the battlecruiser objective
Fixes the same player being able to reroll battlecruiser

## Why It's Good For The Game

The codeowner has stated he doesnt like the check because final objectives should not have volatile checks that can easily change within 10-15 minutes.

## Changelog

:cl:
balance: Removes the ghost check for the battlecruiser objective to appear
fix: Same player cant reroll battlecruiser multiple times anymore
/:cl:

### CODE REVIEW

**Review by Watermelon914 — APPROVED** [review]

**Comment by private-tristan** [comment]
#68057 tried doing this and it was shot down because 'it needed to feel like a full team' or something like that, even though I have played since progtot came out and still have never seen the battlecruiser objective. 
image

**Comment by Watermelon914** [comment]
> #68057 tried doing this and it was shot down because 'it needed to feel like a full team' or something like that, even though I have played since progtot came out and still have never seen the battlecruiser objective.

It was a bad solution, it should've outright removed the check. I don't support these kind of volatile checks for final objectives because you can only get a final objective once.

**Comment by Mothblocks** [comment]
CC @optimumtact

File: code/modules/shuttle/battlecruiser_starfury.dm [hunk 1]
```diff
@@ -171,7 +172,10 @@
 			spawner.antag_team = team
 			if(candidates.len > 0)
 				var/mob/our_candidate = candidates[1]
+				if(excluded_candidates.Find(our_candidate.ckey))
+					continue
 				spawner.create(our_candidate)
+				excluded_candidates += our_candidate.ckey
 				candidates.Splice(1, 2)
```
[the remark this answers is not in the document]
Comment (Salex08) [reply code/modules/shuttle/battlecruiser_starfury.dm:179]: I think its to shorten the list

[the remark this answers is not in the document]
Comment (Salex08) [reply code/modules/shuttle/battlecruiser_starfury.dm:179]: ahh yeah good that you point it out I forgot to add it in the if clause

[the remark this answers is not in the document]
Comment (Salex08) [reply code/modules/shuttle/battlecruiser_starfury.dm]: ok fixed it much better

[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/shuttle/battlecruiser_starfury.dm]: You need to edit and add custom behaviour to the ghost roles that spawn when the roles aren't completely allocated out to every ghost.

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/modules/mob_spawn/ghost_roles/space_roles.dm [hunk 2]
```diff
@@ -117,12 +117,19 @@
 	/// The antag datum to give to the player spawned
 	var/antag_datum_to_give = /datum/antagonist/battlecruiser
 
+/obj/effect/mob_spawn/ghost_role/human/syndicate/battlecruiser/allow_spawn(mob/user, silent = FALSE)
+	if(!(user.key in antag_team.players_spawned))//one per person unless you get a bonus spawn
```
[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/mob_spawn/ghost_roles/space_roles.dm:121]: you may also want to consider including players from the poll in here

[the remark this answers is not in the document]
Comment (Salex08) [reply code/modules/mob_spawn/ghost_roles/space_roles.dm:121]: I thought they get added here
grafik

[the remark this answers is not in the document]
Comment (Salex08) [reply code/modules/mob_spawn/ghost_roles/space_roles.dm:121]: alright done

**Comment by Mothblocks** [comment]
pls stop force pushing if you can help it it makes the notifications maints get harder to interpret

**Comment by Salex08** [comment]
> pls stop force pushing if you can help it it makes the notifications maints get harder to interpret

I forgor to sqash

**Comment by Mothblocks** [comment]
you dont need to squash we squash on merge

**Pull request merged**